### DISCUSSION TITLE
How get Active Locale URL

### DISCUSSION THREAD

**Discussion by oim37** [wintercms/wn-translate-plugin#49] [Q&A] [answered]
Hello,

Help please. I have 3 languages
domain.tld/ - (EN) by default, In plugin config i setup 'prefixDefaultLocale' => env('TRANSLATE_PREFIX_LOCALE', false) to disable en prefix in default
and
domain.tld/es
domain.tld/fr

Previously, I used this configuration in the link

href="/{{ activeLocale }}/page123-my"

1. But now if language EN i need to get URL without prefix in my template. How i can make this, i need - active Locale Url , as I understand it is not have.

2. If i go to domain.tld/en/my-page it will be redirecting to domain.tld/my-page  - I think it can be done in nginx, but I decided to clarify, maybe there is an option in the settings .

Thank you.

**Comment by mjauvin** [comment, marked as answer]
just use the page filter like this:

```
href="{{ 'my-page' | page }}"
```

**Comment by oim37** [comment]
It working only with CMS - Pages, some pages i use with module static pages and it not working both. May be have universal solution or how use filter with static pages module. If CMS page no have use href="{{ 'my-page' | staticPage }}" how it make

**Comment by mjauvin** [comment]
for static pages, there's a special filter "staticPage". See Winter.Pages plugin docs

**Comment by oim37** [comment]
I think I understand, thank you very much.

**Comment by oim37** [comment]
One more question. I have news block with page

/news/:slug 

In language select i use this links to return user on same page with another language
href="/es{{ this.page.url }}
href="/fr{{ this.page.url }}

when i'm on the page domain.tld/news/happy-2023 and want to swith language i see

domain.tld/es/news/:slug 
but i need 
domain.tld/es/news/happy-2023

I try use href="#" data-request="onSwitchLocale" data-request-data="locale: 'en' what in docs, but it not work because i think turn off prefixDefaultLocale. How get right link in {{ this.page.url }} with :slug page. All other working good.

**Comment by mjauvin** [comment]
The localePicker component will do this:
```
{% component "localePicker" %}
```

Otherwise, you can do so manually:
```
{% for code, name in locales | filter((name,code) => code != activeLocale) %}
  <li class="locale-selector position-relative nav-item my-2 white-fg black-bg">
        <a href data-request="onSwitchLocale" data-request-data="locale:'{{ code }}'">{{ code|upper }}</a>
  </li>
{% endfor %}
```

**Comment by oim37** [comment]
I try, it working only if ( plugin config settings )
'prefixDefaultLocale' => env('TRANSLATE_PREFIX_LOCALE', true)
if this value setup false
'prefixDefaultLocale' => env('TRANSLATE_PREFIX_LOCALE', false)

language not changing. On swith, page reloaded with same language.
